src/packet.ml

```diff
@@ -430,25 +427,20 @@ module Tls_crypt = struct
     ({ local_session; packet_id; timestamp; hmac }, clear_hdr_len)
 end
 
-type pkt =
-  [ `Ack of header | `Control of operation * control | `Data of Cstruct.t ]
+type pkt_ack = [ `Ack of header ]
+type pkt_control = [ `Control of operation * control ]
+type pkt = [ pkt_ack | pkt_control | `Data of Cstruct.t ]
+type 'a t = int * ([< pkt ] as 'a)
```

Comment:
what about naming it `type ack` and `type control`?

Reply:
and I have not checked it, but the `type pkt` and `type 'a t` -- can we get rid of one of them? AFAICT `Packet.t` is only refered to once, in `` Bad_mac of t * Cstruct.t * Packet.t`... maybe worth to remove it, rename `type pkt` above to `type t`, and have a `int * Packet.t` where needed?

Reply:
`type control` already existed, but it was only used once so I inlined it as well as `pp_control`. I pushed a commit addressing your comments. I kept `type t` but without the type variable.